OrbitService/Utils.cpp

```diff
@@ -175,29 +158,112 @@ ErrorMessageOr<std::vector<orbit_grpc_protos::TracepointInfo>> ReadTracepoints()
   return result;
 }
 
-ErrorMessageOr<std::unordered_map<pid_t, double>> GetCpuUtilization() {
-  const std::string cmd = "top -b -n 1 -w512 | sed -n '8, 1000{s/^ *//;s/ *$//;s/  */,/gp;};1000q'";
-  OUTCOME_TRY(top_data, ExecuteCommand(cmd));
-  return GetCpuUtilization(top_data);
+static std::optional<pid_t> ProcEntryToPid(const std::filesystem::directory_entry& entry) {
+  if (!entry.is_directory()) {
+    return std::nullopt;
+  }
+
+  int potential_pid;
+  if (!absl::SimpleAtoi(entry.path().filename().string(), &potential_pid)) {
+    return std::nullopt;
+  }
+
+  if (potential_pid <= 0) {
+    return std::nullopt;
+  }
+
+  return static_cast<pid_t>(potential_pid);
+}
+
+std::vector<pid_t> GetAllPids() {
+  const auto emplace_back_if_has_value = [](auto vec, const auto& val) {
+    if (val.has_value()) {
+      vec.emplace_back(std::move(val.value()));
+    }
+
+    return std::move(vec);
+  };
+
+  fs::directory_iterator proc{"/proc"};
+  return std::transform_reduce(begin(proc), end(proc), std::vector<pid_t>{},
+                               emplace_back_if_has_value, ProcEntryToPid);
+}
+
+std::optional<Jiffies> GetCumulativeCpuTimeFromProcess(pid_t pid) {
+  const auto stat = std::filesystem::path{"/proc"} / std::to_string(pid) / "stat";
```

Comment:
add comment with format/example of /stat so this is clear which part this function is extracting.

Reply:
Done.